java-squid/src/main/java/org/sonar/java/SonarComponents.java

```diff
@@ -52,23 +53,26 @@
 
 public class SonarComponents implements BatchExtension {
 
+  private static final boolean IS_SONARQUBE_52 = isSonarQube52();
+
   private final FileLinesContextFactory fileLinesContextFactory;
   private final ResourcePerspectives resourcePerspectives;
   private final JavaTestClasspath javaTestClasspath;
   private final CheckFactory checkFactory;
+  private final SensorContext context;
   private final FileSystem fs;
   private final JavaClasspath javaClasspath;
   private final List<Checks<JavaCheck>> checks;
   private final List<Checks<JavaCheck>> testChecks;
 
   public SonarComponents(FileLinesContextFactory fileLinesContextFactory, ResourcePerspectives resourcePerspectives, FileSystem fs,
-    JavaClasspath javaClasspath, JavaTestClasspath javaTestClasspath,
+    JavaClasspath javaClasspath, JavaTestClasspath javaTestClasspath, SensorContext context,
     CheckFactory checkFactory) {
-    this(fileLinesContextFactory, resourcePerspectives, fs, javaClasspath, javaTestClasspath, checkFactory, null);
+    this(fileLinesContextFactory, resourcePerspectives, fs, javaClasspath, javaTestClasspath, checkFactory, context, null);
   }
 
   public SonarComponents(FileLinesContextFactory fileLinesContextFactory, ResourcePerspectives resourcePerspectives, FileSystem fs,
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Constructor has 8 parameters, which is greater than 7 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS00107)